Examples/Java/Sources/Board.java

```diff
@@ -21,42 +20,289 @@
+        return this.counts;
+    }
+    public @Nullable Date getCreatedAt() {
+        return this.createdAt;
+    }
+    public @Nullable Map<String, String> getCreator() {
+        return this.creator;
+    }
+    public @Nullable String getDescriptionText() {
+        return this.descriptionText;
+    }
+    public @NonNull Image getImage() {
+        return this.image;
+    }
+    public @Nullable String getName() {
+        return this.name;
+    }
+    public @Nullable String getUrl() {
+        return this.url;
+    }
+    public boolean getIdentifierIsSet() {
+        return (this._bits & ID_SET) == ID_SET;
+    }
+    public boolean getContributorsIsSet() {
+        return (this._bits & CONTRIBUTORS_SET) == CONTRIBUTORS_SET;
+    }
+    public boolean getCountsIsSet() {
+        return (this._bits & COUNTS_SET) == COUNTS_SET;
+    }
+    public boolean getCreatedAtIsSet() {
+        return (this._bits & CREATED_AT_SET) == CREATED_AT_SET;
+    }
+    public boolean getCreatorIsSet() {
+        return (this._bits & CREATOR_SET) == CREATOR_SET;
+    }
+    public boolean getDescriptionTextIsSet() {
+        return (this._bits & DESCRIPTION_SET) == DESCRIPTION_SET;
+    }
+    public boolean getImageIsSet() {
+        return (this._bits & IMAGE_SET) == IMAGE_SET;
+    }
+    public boolean getNameIsSet() {
+        return (this._bits & NAME_SET) == NAME_SET;
+    }
+    public boolean getUrlIsSet() {
+        return (this._bits & URL_SET) == URL_SET;
+    }
+    public static class Builder {
     
-        public abstract Builder setIdentifier(@Nullable String value);
-        public abstract Builder setContributors(@Nullable Set<User> value);
-        public abstract Builder setCounts(@Nullable Map<String, Integer> value);
-        public abstract Builder setCreatedAt(@Nullable Date value);
-        public abstract Builder setCreator(@Nullable Map<String, String> value);
-        public abstract Builder setDescriptionText(@Nullable String value);
-        public abstract Builder setImage(@NonNull Image value);
-        public abstract Builder setName(@Nullable String value);
-        public abstract Builder setUrl(@Nullable String value);
-        public abstract Board build();
+        @SerializedName("id") private @Nullable String identifier;
+        @SerializedName("contributors") private @Nullable Set<User> contributors;
+        @SerializedName("counts") private @Nullable Map<String, Integer> counts;
+        @SerializedName("created_at") private @Nullable Date createdAt;
+        @SerializedName("creator") private @Nullable Map<String, String> creator;
+        @SerializedName("description") private @Nullable String descriptionText;
+        @SerializedName("image") private @NonNull Image image;
+        @SerializedName("name") private @Nullable String name;
+        @SerializedName("url") private @Nullable String url;
+         private int _bits = 0;
+        private Builder() {
```

Comment:
Do we need an empty builder init here?

Reply:
It's not a hard requirement right now (creating a Model from scratch), but I'm about to introduce a custom GSON TypeAdapter that goes through the Builder (so that we properly set _bits), and it will use this.

Reply:
Is there a reason we can’t generate the type adapter as well in a separate file?

Reply:
Is there anything proprietary about this adapter? If it’s pretty general it would be useful to include here for others to utilize.

Reply:
@RicoYao - We can resolve this in a follow-up. I'll file an issue

Reply:
@rahul-malik Sorry I should have been more clear, the plan is to submit the TypeAdapter into the master branch (working on the pull request).